resource-managers/kubernetes/core/src/main/scala/org/apache/spark/shuffle/KubernetesLocalDiskShuffleExecutorComponents.scala

```diff
@@ -80,7 +84,7 @@ object KubernetesLocalDiskShuffleExecutorComponents extends Logging {
           .flatMap(_.listFiles).filter(_.isDirectory) // executor-xxx
           .flatMap(_.listFiles).filter(_.isDirectory) // blockmgr-xxx
           .flatMap(_.listFiles).filter(_.isDirectory) // 00
-          .flatMap(_.listFiles)
+          .flatMap(_.listFiles).filterNot(_.getName.contains(".checksum"))
```

Comment:
Hmm, don't we need checksum files?

Reply:
I remember we will use checksum files when reading shuffle data.

Reply:
This is a recovery logic from the PVC of the previously dead executors. The checksum files are optional and currently it's already ignored already.

Reply:
Here is the current logic where `UnrecognizedBlockId` happens for the checksum files.

- https://github.com/apache/spark/blob/57b985c5d57eea614024a4e125eba63663978de3/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/shuffle/KubernetesLocalDiskShuffleExecutorComponents.scala#L99

Reply:
Oh, I see. Thanks. Then checksum files are ignored immediately.